buildSrc/src/main/groovy/org/gradle/build/docs/UserGuideSectionVerifierTask.groovy

```diff
@@ -0,0 +1,97 @@
+package org.gradle.build.docs
+
+import groovy.transform.CompileStatic
+import groovy.transform.ToString
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.TaskAction
+import org.w3c.dom.NodeList
+
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.xpath.XPathConstants
+import javax.xml.xpath.XPathFactory
+
+
+/**
+ * Created by pledbrook on 03/05/2016.
+ */
+@CompileStatic
+class UserGuideSectionVerifierTask extends DefaultTask {
```

Comment:
I'd love to see a test for the custom task e.g. implemented via `ProjectBuilder`. The test would create a new instance of a project, create a task of this type and then execute the action by calling `execute()`. The test should cover all possible use cases.